### PR TITLE
docs(events): Fuller example with pub/sub

### PR DESCRIPTION
**Pull request by kamranayub**
===:clipboard: PR Checklist :clipboard:===

- [ ] :pushpin: issue exists in github for these changes
- [x] :microscope: existing tests still pass
- [x] :see_no_evil: code conforms to the [style guide](https://github.com/excaliburjs/Excalibur/blob/main/STYLEGUIDE.md)
- [ ] :triangular_ruler: new tests written and passing / old tests updated with new scenario(s)
- [x] :page_facing_up: changelog entry added (or not needed)

==================

## Preview

https://kamranayub-docs-events-pubsu.excaliburjs.pages.dev/docs/events#pubsub-or-signal-based-event-bus

## Changes

- Add a Shiki Twoslash example implementing custom event handling
- Show how to achieve a pub/sub emitter
- Show how to achieve higher strictness on event name handling
- Clean up existing verbiage and add some more context

## Notes

- Inspired by @jyoung4242's [post](https://dev.to/excaliburjs/alien-waves-148m) about Signals and Pub/Sub, I noticed our docs lacked info about this use case. When I dug into it, the existing example was not enough to help me create a custom event so I had to look at the engine source.

### CODE REVIEW

**Comment by kamranayub** [comment]
OK, so if I create my own strict emitter, and override the overloads, it will work:

image

I could include this example, although it _feels like_ Ex could provide this or support this somehow in an easier way...

File: site/docs/02-fundamentals/04-events.mdx [hunk 1]
```diff
@@ -17,23 +18,85 @@ Excalibur events are handled synchronously, which is great for debugging and red
...
+
+// ---cut---
+export class Player extends ex.Actor {
+  public events = new ex.EventEmitter<ex.ActorEvents & PlayerEvents>();
+  private health: number = 100;
+
+  public onPostUpdate() {
+    if (this.health <= 0) {
+      // There is a typo in the event name
+      this.events.emit("healthdpleted", new PlayerHealthDepletedEvent(this));
```
[the remark this answers is not in the document]
Comment (kamranayub) [reply site/docs/02-fundamentals/04-events.mdx]: Solved with a custom emitter, but not sure if we want to advise it.

**Comment by kamranayub** [comment]
I'm giving myself credit for Hacktoberfest 🎃

**Review by eonarheim — APPROVED** [review]

**Comment by eonarheim** [comment]
@kamranayub Thanks for the add this helps clear it up a ton!

**Pull request merged**